### ISSUE TITLE
Auto select child items

### ISSUE THREAD

**Issue by nickvanhooydonk** [pnp/sp-dev-fx-property-controls#245]
#### Category
- [x] Enhancement
- [ ] Bug
- [ ] Question

#### Desired Behavior
It would be great if we could have the possibility to auto-select all child items when selecting a parent item (and vice versa on deselecting). 
In addition to this it would be even more super if this could be a three-state selection visual (selected | not selected | partially selected). Example:

![image](https://user-images.githubusercontent.com/17179275/81193254-70106380-8fbb-11ea-9bb8-422190300fb2.png)

**Comment by AJIXuMuK** [comment]
Hi @nickvanhooydonk! What control are you referencing? We don't have `TreeView` yet...

**Comment by nickvanhooydonk** [comment]
Hi @AJIXuMuK,
A bit stupid, but I just found out I posted this in the wrong repository.

Nevertheless, I think adding the treeview from [@pnp/spfx-controls-react](https://pnp.github.io/sp-dev-fx-controls-react/controls/TreeView/) to the pnp property controls will give developers many new opportunities.

What I am missing in the treeview controll is the three-state selection as you see in the screen above. This is a better visiualisation for the end users when a node with subnodes selected is collabsed.

I will check the other repo and request my ticket there.
